tests/robustness/linearizability_test.go

```diff
@@ -203,18 +203,21 @@ func testRobustness(ctx context.Context, t *testing.T, lg *zap.Logger, config e2
 func runScenario(ctx context.Context, t *testing.T, lg *zap.Logger, clus *e2e.EtcdProcessCluster, traffic trafficConfig, failpoint FailpointConfig) (operations []porcupine.Operation, responses [][]watchResponse) {
 	// Run multiple test components (traffic, failpoints, etc) in parallel and use canceling context to propagate stop signal.
 	g := errgroup.Group{}
+
 	trafficCtx, trafficCancel := context.WithCancel(ctx)
 	g.Go(func() error {
+		defer trafficCancel()
 		triggerFailpoints(ctx, t, lg, clus, failpoint)
 		time.Sleep(time.Second)
-		trafficCancel()
 		return nil
 	})
 	maxRevisionChan := make(chan int64, 1)
 	g.Go(func() error {
+		defer func() {
+			maxRevisionChan <- operationsMaxRevision(operations)
```

Comment:
Just curiosity, the defer is used to recover just in case that `simulateTraffic` panic?

Reply:
That was the intention, however I don't think it's correct. I will revert this line

Reply:
Yes, it would not work without a change I just made. I have updated `operationsMaxRevision` to always correct revision which is at least 1.